### ISSUE TITLE
Setting "unique: false" in a schema seems to automatically imply "index: true"

### ISSUE THREAD

**Issue by sean-daley** [Automattic/mongoose#10738]
**Do you want to request a *feature* or report a *bug*?**
Report a potential bug

**What is the current behavior?**
If I set "unique: false" on an attribute in a schema, the collection that gets created is automatically getting an index for that attribute. Even when I do not set "index: true"

**If the current behavior is a bug, please provide the steps to reproduce.**
```
const mongoose = require("mongoose");

const { Schema } = mongoose;

const userSchema = new Schema({
  email: {
    type: Schema.Types.String,
    required: true
  },
  username: {
    type: Schema.Types.String,
    unique: false,
    required: true
  },
});

const User = mongoose.model("User", userSchema);

const main = async () => {
   const client = await mongoose.connect('mongodb://localhost:27017/test', {
  connectTimeoutMS: 3000
});
  const user = new User({
    username: "username",
    email: "user@example.com",
  });

  const dbUser = await user.save();
  console.log(dbUser.toObject());
  client.disconnect();
}

main();
```

If you do a `db.users.getIndices()` after doing this you will see a regular index on username.
If you change the schema by either removing `unique: false` or adding `index: false` then the index goes away.

**What is the expected behavior?**
It would seem like just having "unique: false" without the other attributes should not generate an index automatically. 

Our primary use for doing this is we have helper methods we use to generate all of our schema attributes. These methods take true/false values for things like trim, unique, lowercase, etc. If we have to, we can try and go and fix our schemas to avoid setting unique at all instead of putting unique: false but it seems like this might just be a bug?

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**
Node: v14.17.2
Mongoose: 6.0.6
Mongodb: 4.2.13

Note, this has been happening for a long while now (so with older versions of mongoose 5.x as well). So this behavior isn't new.

**Comment by IslandRhythms** [comment]
https://mongoosejs.com/docs/api.html#schematypeoptions_SchemaTypeOptions-unique

**Comment by sean-daley** [comment]
> https://mongoosejs.com/docs/api.html#schematypeoptions_SchemaTypeOptions-unique

I've read that documentation but it says "when truthy it will build a unique index on ...". I'm not passing in a truthy value though.

**Comment by IslandRhythms** [comment]
My bad I misread your code

**Comment by IslandRhythms** [comment]
Val, the index presence alternates on every run
```javascript
const mongoose = require("mongoose");

const { Schema } = mongoose;

const userSchema = new Schema({
  email: {
    type: Schema.Types.String,
    required: true
  },
  username: {
    type: Schema.Types.String,
    unique: false,
    required: true
  },
});

const User = mongoose.model("User", userSchema);

const main = async () => {
   const client = await mongoose.connect('mongodb://localhost:27017/test', {
  connectTimeoutMS: 3000
});
await mongoose.connection.dropDatabase();
  const user = new User({
    username: "username",
    email: "user@example.com",
  });

  const dbUser = await user.save();
  console.log(dbUser.toObject());
  console.log(await User.listIndexes())
  client.disconnect();
}

main();
```

**Comment by vkarpov15** [comment]
Fix will be in v6.0.8.

For posterity, this issue is closely related to #7620
